src/main/scala/mesosphere/marathon/core/task/tracker/impl/TaskSerializer.scala

```diff
@@ -138,6 +149,24 @@ object TaskSerializer {
   }
 }
 
+object MarathonTaskStatusSerializer {
+  def fromProto(proto: Protos.MarathonTask.MarathonTaskStatus): MarathonTaskStatus = {
+    val result = MarathonTaskStatus.all.collectFirst {
+      case status: MarathonTaskStatus if status.toString == "" + proto => status
+    }
+    // TODO ju<>me discuss
+    result.orNull
+  }
+
+  def toProto(marathonTaskStatus: MarathonTaskStatus): Protos.MarathonTask.MarathonTaskStatus = {
+    val result = Protos.MarathonTask.MarathonTaskStatus.values().collectFirst {
+      case status: Protos.MarathonTask.MarathonTaskStatus if status.toString == "" + marathonTaskStatus => status
+    }
+    // TODO ju<>me discuss
+    result.orNull
```

Comment:
`MarathonTaskStatus` is a sealed trait, therefore it's compile safe to do a pattern match here